salt/modules/debbuild.py

```diff
@@ -395,18 +397,42 @@ def build(runas,
                 log.error('Error building from {0}: {1}'.format(dsc, exc))
             finally:
                 shutil.rmtree(results_dir)
+
     shutil.rmtree(dsc_dir)
     return ret
 
 
-def make_repo(repodir, keyid=None, env=None):
+def make_repo(repodir, keyid=None, env=None, use_passphrase=False, gnupghome='/etc/salt/gpgkeys', runas='root'):
     '''
     Given the repodir, create a Debian repository out of the dsc therein
 
     CLI Example::
 
         salt '*' pkgbuild.make_repo /var/www/html/
+
+    repodir
+        The directory to find packages that will be in the repository
+
+    keyid
+        Optional Key ID to use in signing repository
+
+    env
+        Optional dictionary of environment variables to be utlilized in creating the repository.
+
+        .. versionadded:: 2016.3.0
+
+    use_passphrase
+        Whether to use a passphrase with the signing key.  Passphrase is received from pillar.
+
+    gnupghome
+        Specify the location where GPG related files are stored.
+
+    runas
+        User to create the repository as
     '''
+    # ensure all product owned by runas
+    __salt__['cmd.run']('chown {0}:{0} -R {1}'.format(runas, repodir))
```

Comment:
I don't like this at all. If permissions are to be modified, they should be done so in a different operation and not as a part of a repo creation and certainly not every time this is run.